### ISSUE TITLE
Functional Endpoints, unexpected path pattern matching with pathVariable

### ISSUE THREAD

**Issue by tka4ukalex** [spring-projects/spring-boot#24520]
Problem : I have 2 functional endpoints

```
@Bean
    public RouterFunction<ServerResponse> getById() {
        return RouterFunctions.route(
                GET("/threads/{threadId}"),
                request -> ServerResponse.ok().bodyValue("/threads/{threadId}")
        );
    }
```

```
    @Bean
    public RouterFunction<ServerResponse> getSomeEntity() {
        return RouterFunctions.route(
                GET("/threads/some"),
                request -> ServerResponse.ok().bodyValue("/threads/some"));
    }
```
    
    
  When i make request with GET /threads/some 
  **Expected behavior** : handler from getSomeEntity router function bean will be invoked, response (/threads/some)
   **Actual behavior** : handler from getById router function bean will be invoked,  response (/threads/{threadId})
   
   so it is impossible to invoke handler function from getSomeEntity() bean, because always invokes getById.
   
   also tried to unite this 2 functions into one RouterFunction bean, problem also reproduces
   
   **NOTE**
   if try to model same situation with @RestController and @GetMapping
   pattern matching works as expected
   
   spring boot version: 2.3.5 Release
   
   from dependencies only 
    <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-webflux</artifactId>
        </dependency>

**Comment by philwebb** [comment]
I don't think that `RouterFunction` beans work in the same way as `@RestController` mappings. I think the first matching route will win. You could try adding `@Order` to your bean definitions so that the `/threads/some` function is before the pattern.

I'll transfer this issue to the Spring Framework team that are responsible for this code.
